src/AggregateFunctions/AggregateFunctionJavaScriptAdapter.cpp

```diff
@@ -206,11 +215,24 @@ JavaScriptAggrFunctionState::~JavaScriptAggrFunctionState()
 
 void JavaScriptAggrFunctionState::add(const IColumn ** src_columns, size_t row_num)
 {
-    for (size_t i = 0; auto & col : columns)
-    {
-        col->insertFrom(*src_columns[i], row_num);
-        i++;
-    }
+    size_t num_of_input_columns = columns.size() - 1;
+
+    for (size_t i = 0; i < num_of_input_columns; i++)
+        columns[i]->insertFrom(*src_columns[i], row_num);
+
+    /// _tp_delta column
+    columns[columns.size() - 1]->insert(1);
```

Comment:
Are we always expecting `_tp_delta` column even for non-changelog scenario ?

```
columns.back()->insert(1);
```

Reply:
Yes, because in the UDA, _tp_delta should always be the last input argument.